src/main/java/baritone/utils/schematic/format/defaults/LitematicaSchematic.java

```diff
@@ -0,0 +1,345 @@
+/*
+ * This file is part of Baritone.
+ *
+ * Baritone is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Baritone is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Baritone.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package baritone.utils.schematic.format.defaults;
+
+import baritone.utils.accessor.INBTTagLongArray;
+import baritone.utils.schematic.StaticSchematic;
+import net.minecraft.block.Block;
+import net.minecraft.block.properties.IProperty;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.Vec3i;
+import org.apache.commons.lang3.Validate;
+
+import javax.annotation.Nullable;
+import java.util.Optional;
+
+/**
+ * @author Emerson
+ * @since 12/27/2020
+ * @author rycbar
+ * @since 22.09.2022
+ */
+public final class LitematicaSchematic extends StaticSchematic {
+    private final Vec3i offsetMinCorner;
+    private final NBTTagCompound nbt;
+
+    /**
+     * @param nbtTagCompound a decompressed file stream aka nbt data.
+     * @param rotated        if the schematic is rotated by 90° aka x and z size are switched.
+     */
+    public LitematicaSchematic(NBTTagCompound nbtTagCompound, boolean rotated) {
+        this.nbt = nbtTagCompound;
+        this.offsetMinCorner = new Vec3i(getMinOfSchematic("x"), getMinOfSchematic("y"), getMinOfSchematic("z"));
+        this.y = Math.abs(nbt.getCompoundTag("Metadata").getCompoundTag("EnclosingSize").getInteger("y"));
+
+        if (rotated) {
```

Comment:
this isn't rotation. if you swap x and z that's actually a mirror along a 45 degree axis. swapping x and z is not the same as a 90 degree rotation. in fact, swapping x and z is not the same as rotating by any number of degrees, it's a mirror not a rotate

Reply:
what i meant here was that the size in the x direction after a 90° rotation is the same as the size in the z direction befor the turn and vice versa

Reply:
ok plz add a comment explaining how rotation versus mirror works so that others arent confused in the same way that i am (because right now it looks wrong to the reader)